--- a/net/minecraft/client/renderer/SubmitNodeStorage.java
+++ b/net/minecraft/client/renderer/SubmitNodeStorage.java
@@ -280,4 +_,10 @@
     @OnlyIn(Dist.CLIENT)
     public record TranslucentModelSubmit<S>(SubmitNodeStorage.ModelSubmit<S> modelSubmit, RenderType renderType, Vector3f position) {
     }
+
+    public void submitCustomRenderPass(PoseStack poseStack, CustomRenderPassRenderer renderer) {
+        order(0).submitCustomRenderPass(poseStack, renderer);
+    }
+
+    public record CustomRenderPassSubmit(PoseStack.Pose pose, CustomRenderPassRenderer renderer) { }
 }
